### PR TITLE
parametrized key-value storage implemented

### PR DESCRIPTION
**Pull request by lavrivskyi**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,50 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_VALUES_QUANTITY = 10;
+    private K[] keys;
+    private V[] values;
+    private int putIndex;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]:
```suggestion
    private int size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,50 @@
...
+            }
+        }
+        keys[putIndex] = key;
+        values[putIndex] = value;
+        putIndex++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < MAX_VALUES_QUANTITY; i++) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:36]: Maybe it's enough to iterate only over the known count of elements, not over all array capacity?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,50 @@
...
     }
 
     @Override
     public int size() {
-        return -1;
+        int size = 0;
+        for (V value : values) {
+            size += value != null ? 1 : 0;
+        }
+        return size;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:50]: please simplify this method. It's enough to return size
```suggestion
        return size;
```

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,48 @@
...
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        int valueIndex = 0;
+        for (K k : keys) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:37]: you can use `for i` loop and in this case you are not required to have `valueIndex` variable

**Review by boroda4436 — CHANGES_REQUESTED** [review]

**Review by Iryna-U — APPROVED** [review]

**Pull request closed without merging**